### DISCUSSION TITLE
download data with filters in mind

### DISCUSSION THREAD

**Discussion by vukeep** [h2oai/wave#1767] [Q&A] [answered]
Hi! I just started studying H2O Wave, I had a question, I output a large data set to a table, the user can filter and sort my data, but when downloading he gets the whole set, without the filters that were applied, is there a way to download the filtered data set??

**Comment by mturoci** [comment, marked as answer]
Hey @vukeep

Wave offers 2 kinds of table:

* client side - simpler, the one that you already use now https://wave.h2o.ai/docs/widgets/form/table/#basic-table - most of the stuff you can get for free, but you are giving up control over the actions like filter, download etc.
* server side (paginated) - this is for cases when you have so much data that the user would never be able to scroll through it all, so you need to add [pagination](https://wave.h2o.ai/docs/widgets/form/table/#with-pagination) and handle all the events like filter, download yourself, then present your user with the data you want. For your particular case, you would want to save the active filters and send the data with the current filter applied when the user hits download. See [this example](https://wave.h2o.ai/docs/examples/table-pagination/) for more info.

**Reply by vukeep** [reply]
Hey @mturoci!
thank you so much for the answer! did I understand correctly that in order to get control over the download, I will have to overload the contents of the table each time in accordance with the imposed user filters?

**Reply by mturoci** [reply]
Yes, that's right. You either have no control (client side table) or full (maybe too much :) ) control - server side table.

**Reply by vukeep** [reply]
@mturoci thanks!!! )
